internal/commands/app/create_inputs_test.go

```diff
@@ -372,111 +373,659 @@ func TestAppCreateInputsResolveDirectory(t *testing.T) {
+		_, ui := mock.NewUI()
 		var expectedGroupID string
 		ac := mock.AtlasClient{}
 		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
 			expectedGroupID = groupID
-			return nil, nil
+			return []atlas.Cluster{
+				{ID: "789", Name: "test-cluster-1"},
+				{ID: "1011", Name: "test-cluster-2"},
+			}, nil
 		}
 
-		inputs := createInputs{Cluster: "test-cluster"}
+		inputs := createInputs{
+			newAppInputs:        newAppInputs{Name: "test-app"},
+			Clusters:            []string{"test-cluster-1", "test-cluster-2"},
+			ClusterServiceNames: []string{"mongodb-atlas", "another-data-source"},
+		}
+
+		ds, _, err := inputs.resolveClusters(ui, ac, "123")
+		assert.Nil(t, err)
 
-		_, err := inputs.resolveCluster(ac, "123")
-		assert.Equal(t, errors.New("failed to find Atlas cluster"), err)
+		assert.Equal(t, []dataSourceCluster{
+			{
+				Name: "mongodb-atlas",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-cluster-1",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			},
+			{
+				Name: "another-data-source",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-cluster-2",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			},
+		}, ds)
 		assert.Equal(t, "123", expectedGroupID)
 	})
 
+	t.Run("should warn user if clusters are not found and auto confirm is set", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		console, ui, err := mock.NewConsoleWithOptions(mock.UIOptions{AutoConfirm: true})
+		assert.Nil(t, err)
+		defer console.Close()
+
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "test-app-abcde",
+			Name:        "test-app",
+		}
+
+		rc := mock.RealmClient{}
+		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			return testApp, nil
+		}
+		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+		rc.TemplatesFn = func() ([]realm.Template, error) {
+			return []realm.Template{}, nil
+		}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{{ID: "789", Name: "test-cluster-1"}}, nil
+		}
+
+		dummyClusters := []string{"test-cluster-dummy-1", "test-cluster-dummy-2"}
+		inputs := createInputs{
+			newAppInputs:        newAppInputs{Name: testApp.Name, Project: testApp.GroupID},
+			Clusters:            []string{"test-cluster-1", dummyClusters[0], dummyClusters[1]},
+			ClusterServiceNames: []string{"mongodb-atlas"},
+		}
+
+		doneCh := make(chan (struct{}))
+		go func() {
+			defer close(doneCh)
+			console.ExpectString(fmt.Sprintf("Note: The following data sources were not linked because they could not be found: %s", strings.Join(dummyClusters[:], ", ")))
```

Comment:
[nit] shouldn't need the `[:]` here, and considering its a test, you don't necessarily _need_ the `fmt.Sprintf` here and could just write the expected string.  your call on that last one though